src/wp-includes/class-wp-scripts.php

```diff
@@ -295,14 +304,34 @@ public function do_item( $handle, $group = false ) {
 		}
 
 		$before_handle = $this->print_inline_script( $handle, 'before', false );
-		$after_handle  = $this->print_inline_script( $handle, 'after', false );
 
 		if ( $before_handle ) {
 			$before_handle = sprintf( "<script%s id='%s-js-before'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $before_handle );
 		}
 
+		// Eligible loading strategies will only be 'async', 'defer', or ''.
+		$strategy     = $this->get_eligible_loading_strategy( $handle );
+		$after        = ( '' === $strategy ) ? 'after' : 'after-standalone';
+		$after_handle = $this->print_inline_script( $handle, $after, false );
+
 		if ( $after_handle ) {
-			$after_handle = sprintf( "<script%s id='%s-js-after'>\n%s\n</script>\n", $this->type_attr, esc_attr( $handle ), $after_handle );
+			$after_handle = sprintf(
+				"<script%1\$s id='%2\$s-js-after'>\n%3\$s\n</script>\n",
+				$this->type_attr,
+				esc_attr( $handle ),
+				$after_handle
+			);
+		}
+		if ( '' !== $strategy ) {
```

Comment:
See above.
```suggestion
		if ( 'defer' === $strategy || 'async' === $strategy ) {
```

Reply:
@felixarntz I've updated this to be more explicit as per your suggestion.